src/routes/donors.js

```diff
@@ -101,18 +101,61 @@ router.get('/sums', function(req, res, next) {
   })
 })
 
-router.put('/updatedonors/:id', (req, res) => {
-  console.log('hello world')
-  db.Donor.update(
-    {
-      email: req.body.email
-    },
-    {
-      where: {
-        id: req.params.id
-      }
+router.get('/edit', (req, res) => {
+  db.PreferredContact.findAll({
+    attributes: ['id', 'description']
+  })
+    .then(result => {
+      res.json(result)
+    })
+    .catch(error => {
+      res.status(500).json(error)
+    })
+})
+
+router.put('/edit/:idNo', (req, res) => {
+  const { idNo: donorId } = req.params
+  const { remarks, preferredContact, dnc } = req.body
+  if (!remarks /* && !preferredContact && !('dnc' in req.body) */) {
+    return res.sendStatus(204).json({
+      message: 'You have not sent any parameters for updating'
+    })
+  }
+  try {
+    // We may or may not need to do this findOne, depending on whether the preferred contact is being updated.
+    let prom = Promise.resolve({})
+    if (preferredContact) {
+      prom = db.PreferredContact.findOne({
```

Comment:
You don’t need this whole findOne. If you see on line 141, at the end you are only using the preferredContact.id, which you already had to begin with (before the findOne). So just use the id from the request body directly.

Reply:
okay removed!